gui.py

```diff
@@ -0,0 +1,246 @@
+                                         bg="#00796B", fg="white", font=("Roboto", 12, "bold"),
+                                         relief="flat", bd=0, height=2, width=20, cursor="hand2")
+        self.clock_in_button.pack(anchor="center", pady=20)
+
+
+
+        # Total hours worked
+        self.total_hours_label = ttk.Label(self.log_in_frame, text="Total hours: 0", font=("Roboto", 12))
+        self.total_hours_label.pack(anchor="center", pady=10)
+
+    def open_edit_employee_window(self):
+        """Opens a new window to edit employee details."""
+        edit_window = tk.Toplevel(self.root)
+        edit_window.title("Edit Employee")
+        edit_window.geometry("400x300")
+
+        # Example in-line editable employee table (simple example)
+        edit_label = ttk.Label(edit_window, text="Edit employee data here.")
+        edit_label.pack(pady=20)
+
+    def open_reports_window(self):
+        """Opens a new window to view and print reports."""
+        # if self.logged_in_user.is _admin()== False:
+            # return with an error message
+
+        report_window = tk.Toplevel(self.root)
+        report_window.title("Reports")
+        report_window.geometry("500x400")
+
+        # Range of dates and employee selection
+        range_label = ttk.Label(report_window, text="Specify Date Range", font=("Roboto", 12))
+        range_label.pack(pady=10)
+        start_date = tk.Entry(report_window, width=20)
+        start_date.pack(pady=5)
+        end_date = tk.Entry(report_window, width=20)
+        end_date.pack(pady=5)
+
+        employee_label = ttk.Label(report_window, text="Select Employee", font=("Roboto", 12))
+        employee_label.pack(pady=10)
+        employee_combobox = ttk.Combobox(report_window, values=["All Employees", "johndoe", "janedoe"])
+        employee_combobox.pack(pady=5)
+
+        # Print button
+        print_button = tk.Button(report_window, text="Print Report", command=self.print_report,
+                                 bg="#00796B", fg="white", font=("Roboto", 12, "bold"), relief="flat", bd=0)
+        print_button.pack(pady=20)
+
+    def print_report(self):
+        """Sends the report to the default printer."""
+        # Print functionality
+        report_text = "This is a sample report. \nTitle: Business Report \nDate Range: xx/xx/xxxx - xx/xx/xxxx \nAdmin: Admin Name"
+
+        # Use os module to open the default print dialog (platform-specific)
+        if os.name == "nt":  # For Windows
+            file_name = "report.txt"
+            with open(file_name, "w") as f:
+                f.write(report_text)
+            os.startfile(file_name, "print")
+        else:
+            messagebox.showinfo("Print Report", "Printing is not supported on this OS.")
+
+    def clock_in(self):
+        """Handles the clock-in process."""
+        username = self.username.get()
+        pin = self.pin.get()
+
+        if username and pin:
```

Comment:
The conditional should get split up into three parts: if the username and pin are in the database, if they aren't in the database (option to add them), and if not enough data was inputted.